Final steps of week 3. Filtering comments

eriyezab opened the pull request.

eriyezab committed:
Add filtering functionality to the doGet method of data.

eriyezab committed:
Merge with DeleteComments branch.

eriyezab committed:
Merge branch 'master' of https://github.com/eriyezab/my-portfolio into AdditionalFeatures

eriyezab committed:
Attend to TODOs to remove print statements.

eriyezab committed:
Change default sort order to descending and fix default values in DataServlet.java

eriyezab committed:
Add missing word to class comment.

eriyezab committed:
Add dependency to pom.xml to allow for use of Users API.

ankitaj224 commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -29,19 +29,25 @@
+    // Get filter parameters
+    int maxComments = getNumComments(request);
+
+    Query query = makeQueryFromParams(request);
```
Could you prepare a query that limits reading maxComments from Datastore vs doing it in memory? Former will be efficient.

eriyezab committed:
Limit comments directly from datastore instead of in memory.

eriyezab committed:
Add link to comment section in navbar of index.html.

eriyezab committed:
Create servlet to check logged in status of user and create class that represents that status.

eriyezab committed:
Remove duplicate dependecy.

eriyezab committed:
Make comment form hidden by default.

eriyezab committed:
Hide form to logged out users and provide log in link or show form and provide log out link to logged in users.

eriyezab committed:
Add email attribute and provide getters for all private attributes.

eriyezab committed:
Modify DataServlet.java to get and store the current user's email address with the comment.

eriyezab committed:
Add functionality so that comments with no name display the email address but if a name is submitted the name will display.

eriyezab committed:
Change attribute name to better represent meaning.

eriyezab committed:
Add sort by email functionality.

eriyezab committed:
Merge branch 'AdditionalFeatures' into Authentication

eriyezab committed:
Add dependecy to pom.xml to allow for use of Cloud Natural language API.

eriyezab committed:
Add score attribute to the Comment class to hold the sentiment score of the message.

eriyezab committed:
Fix spelling in comment for score attribute.

eriyezab committed:
Add sentiment analysis functionality to comments and display the sentiment score with the comment.

eriyezab committed:
Attend to todo to remove comment.

eriyezab committed:
Merge branch 'master' into AdditionalFeatures

ankitaj224 reviewed (approved):
Good job!

eriyezab committed:
Update getter method name in Comment.java.

eriyezab committed:
Update class javadoc and change attributand getter name to better reflect functionality.

eriyezab committed:
Properly punctuate all comments and javadocs.

eriyezab committed:
Update constant names to be more descriptive and replace assignment with ternary operator.

eriyezab committed:
Update the getUserStatus to reflect the changes made in the UserStatus.java class.

eriyezab committed:
Change attribute name to be more accurate.

eriyezab committed:
Handle exception thrown by sentiment analysis api and remove comments with low score.

eriyezab committed:
Update Comment.java

Remove unintentional double blank lines at end of file

eriyezab committed:
Update DataServlet.java

Update all if, else if, and for statements to be followed by a space.

eriyezab committed:
Update comments.js

Update all if, else if, and for statements to be followed by a space.

eriyezab committed:
Update DataServlet.java

Move stack trace into single log statement

eriyezab committed:
Update comments.js

Use switch statement instead of ternary operation for message and move into its own function for improved readability and in case more reasons are added in the future.

eriyezab committed:
Merge pull request #20 from eriyezab/SentimentAnalysis

Sentiment Analysis API Integration

eriyezab committed:
Merge branch 'AdditionalFeatures' into Authentication

eriyezab committed:
Merge pull request #19 from eriyezab/Authentication

Authentication API Integration
